src/ucp/dt/datatype_iter.inl

```diff
@@ -393,7 +393,7 @@ ucp_datatype_iter_next_slice(const ucp_datatype_iter_t *dt_iter,
 static UCS_F_ALWAYS_INLINE uct_mem_h
 ucp_datatype_iter_uct_memh(const ucp_mem_h memh, ucp_rsc_index_t memh_index)
 {
-    if (memh_index == UCP_NULL_RESOURCE) {
+    if ((memh_index == UCP_NULL_RESOURCE) || (memh == NULL)) {
```

Comment:
can memh be NULL only for IOV type?
if yes, better move memh==NULL check to caller function, to not use it in contig type